app/js/referenceEdit.controller.js

```diff
@@ -0,0 +1,55 @@
+angular
+    .module('conceptDictionaryApp')
+    .controller('ReferenceEditController', ReferenceEditController);
+
+ReferenceEditController.$inject = ['reference', 'sources', 'openmrsRest', '$location'];
+
+function ReferenceEditController (reference, sources, openmrsRest, $location ){
+
+    var vm = this;
+
+    vm.reference = reference;
+    vm.sources = sources;
+    vm.selectedConceptSource = vm.reference.conceptSource.name;
+
+    vm.cancel = cancel;
+    vm.save = save;
+    vm.deleteForever = deleteForever;
+    vm.retire = retire;
+    vm.updateConceptSource = updateConceptSource;
+
+
+    function updateConceptSource() {
```

Comment:
Please add some comment why we need it i.e. explain that ng-options doesn't display previous value if you work with objects and you must use strings instead.